app/uk/gov/hmrc/brm/connectors/BirthConnector.scala

```diff
@@ -16,98 +16,118 @@
 package uk.gov.hmrc.brm.connectors
 
-import java.net.URLEncoder
-
-import play.api.Logger
-import play.api.Play.current
-import play.api.libs.ws.WS
-import uk.gov.hmrc.brm.audit.{BRMAudit, EnglandAndWalesAuditEvent, NorthernIrelandAuditEvent, ScotlandAuditEvent}
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.brm.audit.{BRMAudit, NorthernIrelandAuditEvent, ScotlandAuditEvent}
 import uk.gov.hmrc.brm.config.WSHttp
 import uk.gov.hmrc.brm.models.brm.Payload
-import uk.gov.hmrc.brm.utils.{BirthRegisterCountry, BrmLogger, Keygenerator, NameFormat}
+import uk.gov.hmrc.brm.utils.{BrmLogger, Keygenerator, NameFormat}
 import uk.gov.hmrc.play.config.ServicesConfig
 import uk.gov.hmrc.play.http._
 
 import scala.concurrent.Future
 
 trait BirthConnector extends ServicesConfig {
 
-  val serviceUrl : String
-  val httpGet : HttpGet
+  val serviceUrl: String
+  val httpGet: HttpGet
+  val httpPost: HttpPost
 
-  val baseUri : String
-  val detailsUri : String
+  val baseUri: String
+  val detailsUri: String
+  val referenceUri: String
 
   abstract class RequestType
+
   case class ReferenceRequest() extends RequestType
+
   case class DetailsRequest() extends RequestType
 
-  private val referenceURI : PartialFunction[Payload, String] = {
-    case Payload(Some(birthReferenceNumber), _, _, _, _) =>
-      s"$detailsUri/$birthReferenceNumber"
+  case class RequestDetail(uri: String, jsonBody: Option[JsValue] = None){
+    val headers = Seq(
+                BrmLogger.BRM_KEY -> Keygenerator.geKey(),
+                "Content-Type" -> "application/json; charset=utf-8")
   }
 
-  private val detailsURI : PartialFunction[Payload, String] = {
+  private val referenceBody: PartialFunction[Payload, (String, JsValue)] = {
+    case Payload(Some(brn), _, _, _, _) =>
+      (referenceUri, Json.parse(
+        s"""
+           |{
+           |  "reference" : "$brn"
+           |}
+         """.stripMargin))
+  }
+
+  private val detailsBody: PartialFunction[Payload, (String, JsValue)] = {
     case Payload(None, f, l, d, _) =>
-      val nameValuePair = Map(
-        "forenames" -> NameFormat(f),
-        "lastname" -> NameFormat(l),
-        "dateofbirth" -> s"$d"
-      )
-
-      val query = nameValuePair.map(pair => pair._1 + "=" + URLEncoder.encode(pair._2, "UTF-8")).mkString("&")
-      detailsUri.concat(s"?$query")
+      (detailsUri, Json.parse(
+        s"""
+           |{
+           | "forenames" : "${NameFormat(f)}",
+           | "lastname" : "${NameFormat(l)}",
+           | "dateofbirth" : "$d"
+           |}
+        """.stripMargin))
   }
 
-  private def request(payload: Payload, operation: RequestType)(implicit hc: HeaderCarrier) = {
+  private def buildRequest(payload: Payload, operation: RequestType): RequestDetail = {
+
     val f = operation match {
-      case ReferenceRequest() => referenceURI
-      case DetailsRequest() => detailsURI
+      case ReferenceRequest() => referenceBody
+      case DetailsRequest() => detailsBody
```

Comment:
What if it doesn't satisfy either PF? I know this would not occur coming from matching but think of it as separate.